jenkins/janitor.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import collections
+import datetime
+import re
+import subprocess
+import sys
+
+
+# PR:
+
+# gke:
+# k8s-jkns-pr-gke
+# k8s-jkns-pr-gci-gke
+
+# gce:
+# k8s-jkns-pr-gce
+# k8s-jkns-pr-gce-etcd3
+# k8s-jkns-pr-gci-gce
+# k8s-jkns-pr-bldr-e2e-gce-fdrtn
+# k8s-jkns-pr-gci-bld-e2e-gce-fd
+# k8s-jkns-pr-kubemark
+# k8s-jkns-pr-gci-kubemark
+# kubernetes-pr-cri-validation
+
+# CI:
+# kubernetes-jenkins
```

Comment:
Yeah a lot of worries to the CI project because of a bunch of cannot-delete stuff